app/cisc/cisc.go

```diff
@@ -0,0 +1,312 @@
+package main
+
+import (
+	"os"
+
+	"encoding/hex"
+
+	"path"
+
+	"io/ioutil"
+	"strings"
+
+	"github.com/dedis/cothority/app/lib/config"
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/services/identity"
+	"gopkg.in/codegangsta/cli.v1"
+)
+
+/*
+Cisc is the Cisc Identity SkipChain to store information in a skipchain and
+being able to retrieve it.
+
+This is only one part of the system - the other part being the cothority that
+holds the skipchain and answers to requests from the cisc-binary.
+*/
+
+var configFile string
+var sshDir string
+var sshConfig string
+var clientApp *identity.Identity
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "SSH keystore client"
+	app.Usage = "Connects to a ssh-keystore-server and updates/changes information"
+	app.Commands = []cli.Command{
+		commandID,
+		commandConfig,
+		commandKeyvalue,
+		commandSSH,
+	}
+	app.Flags = []cli.Flag{
+		cli.IntFlag{
+			Name:  "debug, d",
+			Value: 1,
+			Usage: "debug-level: 1 for terse, 5 for maximal",
+		},
+		cli.StringFlag{
+			Name:  "config, c",
+			Value: "~/.cisc",
+			Usage: "The configuration-directory of cisc",
+		},
+		cli.StringFlag{
+			Name:  "config-ssh, cs",
+			Value: "~/.ssh",
+			Usage: "The configuration-directory of the ssh-directory",
+		},
+	}
+	app.Before = func(c *cli.Context) error {
+		configDir := config.TildeToHome(c.String("config"))
+		os.Mkdir(configDir, 0660)
+		log.SetDebugVisible(c.Int("debug"))
+		configFile = configDir + "/config.bin"
+		if err := loadConfig(); err != nil {
+			log.Error("Problems reading config-file. Most probably you\n",
+				"should start a new one by running with the 'setup'\n",
+				"argument.")
+		}
+		sshDir = config.TildeToHome(c.String("cs"))
+		sshConfig = sshDir + "/config"
+		return nil
+	}
+	app.After = func(c *cli.Context) error {
+		if clientApp != nil {
+			err := saveConfig()
+			log.ErrFatal(err, "Error while creating config-file", configFile)
+		}
+		return nil
+	}
+	app.Run(os.Args)
+
+}
+
+/*
+ * Identity-related commands
+ */
+func idCreate(c *cli.Context) error {
+	log.Info("Creating id")
+	if c.NArg() == 0 {
+		log.Fatal("Please give at least a group-definition")
+	}
+
+	group := getGroup(c)
+
+	name, err := os.Hostname()
+	log.ErrFatal(err)
+	if c.NArg() > 1 {
+		name = c.Args().Get(1)
+	}
+	log.Info("Creating new blockchain-identity for", name)
+
+	clientApp = identity.NewIdentity(group.Roster, 2, name)
+	log.ErrFatal(clientApp.CreateIdentity())
+	err = saveConfig()
+	log.ErrFatal(err)
+	return nil
+}
+
+func idConnect(c *cli.Context) error {
+	log.Info("Connecting")
+	name, err := os.Hostname()
+	log.ErrFatal(err)
+	switch c.NArg() {
+	case 2:
+		// We'll get all arguments after
+	case 3:
+		name = c.Args().Get(2)
```

Comment:
Yes, because the arguments are:
`group id [id-name]`
so the `id-name` can be omitted. But as two star-programmers get it wrong, I should perhaps replace that with
`if c.NArg() != 2 && c.NArg() != 3{ log.Fatal("wrong number of arguments) }`
or something like that...